### ISSUE TITLE
Russia

### ISSUE THREAD

**Issue by tavurth** [zhukov/webogram#1671]
Any plans to implement something similar to the current telegram VPN system to allow Russian clients to connect via https://web.telegram.org?

**Comment by hooddy** [comment]
Как вы себе представляете, чтобы приложение в браузере перекрывало настройки браузера?

**Comment by tavurth** [comment]
It seems like the current DNS resolution for `https://web.telegram.org` => `149.154.167.120` has been blocked. Getting a couple of Aliases up and running would perhaps get that milestone out of the way. [example](https://zhukov.github.io/webogram)

As for the client connections, perhaps this is something which could be brought up with TG directly? The web application shown here is an ideal entry point for users. Since the Russian user-base is so large, it would make sense to further support this. 

@hooddy Is there a reason you're so upset?

**Comment by hooddy** [comment]
@tavurth sorry I don't. Web apps get settings from browser and os. So how do they can avoid it and use any other settings? I think this cant be done.

**Comment by dddpaul** [comment]
Hi.

I live in Russia and access to *.telegram.org is blocked for me. 
So I deployed webogram in US hosting and have access to web-interface with my browser.

But when I'm trying to sign in my browser sends request to `https://venus.web.telegram.org/apiw1` which is blocked too :)

The obvious solution is to proxy these requests through my server but is it real?

**Comment by juliogonzalez** [comment]
@dddpaul, that depends. If we are talking about a shared web hosting, then it is not possible.

But if you own the server, then you could install a Proxy (SQUID for example, make sure you secure it), and use it in your browser (for all traffic or only for *.telegram.org) or -of course- you can install openVPN on the server.

An alternative to the US server (if it is shared hosting) is having a VPS (Virtual Private Server) with a very well known french hosting company (I don't want to spam so I will not write the name) for ~3€/month (it should be enough for most cases).

**Comment by NikKovIos** [comment]
I've used Franz with web version of telegram. Write this just in case )

**Comment by stek29** [comment]
#1673

**Comment by penartur** [comment]
@dddpaul 

> The obvious solution is to proxy these requests through my server but is it real?

It was real. I've set up webogram fork on my own domain (behind cloudflare), and changed API endpoints from venus.web.telegram.org etc to the subdomains of my own domain (proxied to real venus.web.telegram.org by cloudflare).

However, it all stopped working a couple of days ago. Now, for all API requests made to my subdomains, with origin headers of my domain, I'm getting 302 permanent redirect to core.telegram.org, without any explanation. I tried to work around this problem, but had no success.
